systemtest/src/main/java/io/strimzi/systemtest/apiclients/MsgCliApiClient.java

```diff
@@ -0,0 +1,233 @@
+                T body = response.body();
+                if (response.statusCode() != expectedCode) {
+                    LOGGER.error("expected-code: {}, response-code: {}, body: {}", expectedCode, response.statusCode(), response.body());
+                    promise.fail(new RuntimeException("Status " + response.statusCode() + " body: " + (body != null ? body.toString() : null)));
+                } else if (response.statusCode() < HTTP_OK || response.statusCode() >= HttpURLConnection.HTTP_MULT_CHOICE) {
+                    promise.fail(new RuntimeException(body.toString()));
+                } else {
+                    promise.complete(ar.result().body());
+                }
+            } else {
+                LOGGER.warn(warnMessage);
+                promise.fail(ar.cause());
+            }
+        } catch (io.vertx.core.json.DecodeException decEx) {
+            if (ar.result().bodyAsString().contains("application is not available")) {
+                LOGGER.warn("'{}' is not available.", apiClientName(), ar.cause());
+                throw new IllegalStateException(String.format("'%s' is not available.", apiClientName()));
+            } else {
+                LOGGER.warn("Unexpected object received", ar.cause());
+                throw new IllegalStateException("JsonObject expected, but following object was received: " + ar.result().bodyAsString());
+            }
+        }
+    }
+
+    /**
+     * Start new messaging kafka client(s)
+     *
+     * @param clientArguments list of arguments for kafka client (together with kafka client name!)
+     * @param count           count of clients that will be started
+     * @return
+     * @throws InterruptedException
+     * @throws ExecutionException
+     * @throws TimeoutException
+     */
+    public JsonObject startClients(List<String> clientArguments, int count) throws Exception {
+        Future<JsonObject> responsePromise = Future.future();
+//        CompletableFuture<JsonObject> responsePromise = new CompletableFuture<>();
+        JsonObject request = new JsonObject();
+        request.put("command", new JsonArray(clientArguments));
+        request.put("count", count);
+
+        return doRequestNTimes(initRetry, () -> {
+                webClient.post(endpoint.getPort(), endpoint.getHost(), "")
+                        .as(BodyCodec.jsonObject())
+                        .timeout(120_000)
+                        .sendJson(request,
+                                ar -> responseHandler(ar, responsePromise, HttpURLConnection.HTTP_OK, "Error starting messaging clients"));
+
+            // Here I should wait till responsePromies will be completed somehow
+            // After putting there breakpoint, responsePromise will never be completed (evn without bp)
+            if (responsePromise.result() == null) {
+                throw new Exception("Result is null!");
+            }
+
+            return responsePromise.result();
```

Comment:
You're returning here before you know whether the `responseHandler()` has been called. The lambda argument to `sendJson()` will be called asynchronously . If you want a blocking lambda to pass to `doRequestNTimes()` then you need to do the blocking yourself. Vertx doesn't provide any nice way to say "block until this `Future` is completed", so you might not want to use the vertx web client at all, but rather some other HTTP client which has the blocking API you seem to want to be using.